lib/model/message.dart

```diff
@@ -35,6 +35,37 @@ mixin MessageStore {
   /// All [Message] objects in the resulting list will be present in
   /// [this.messages].
   void reconcileMessages(List<Message> messages);
+
+  /// Whether the current edit request, if any, has failed.
+  ///
+  /// Will be null if there is no current edit request
+  /// or the message was deleted.
```

Comment:
I think I'd want to conceptualize "the message was deleted" as one way to (re-)enter the state of there being no current edit request. If the message was deleted, then any edit request for it that had been outstanding no longer is.